cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CaReconcilerTest.java

```diff
@@ -1429,220 +1419,1101 @@ public void testClientsCASecretsWithoutOwnerReference(Vertx vertx, VertxTestCont
+                .withNewMetadata()
+                    .withName(NAME + "-broker")
                 .endMetadata()
+                .withNewSpec()
+                    .withPods(PodSetUtils.podsToMaps(brokerPods))
+                .endSpec()
+                .build();
+
+        when(spsOps.listAsync(eq(NAMESPACE), any(Labels.class))).thenReturn(Future.succeededFuture(List.of(controllerPodSet, brokerPodSet)));
+
+        Map<String, Deployment> deps = new HashMap<>();
+        deps.put("my-cluster-entity-operator", deploymentWithName("my-cluster-entity-operator"));
+        deps.put("my-cluster-cruise-control", deploymentWithName("my-cluster-cruise-control"));
+        deps.put("my-cluster-kafka-exporter", deploymentWithName("my-cluster-kafka-exporter"));
+        DeploymentOperator depsOperator = supplier.deploymentOperations;
+        when(depsOperator.getAsync(any(), any())).thenAnswer(i -> Future.succeededFuture(deps.get(i.getArgument(1, String.class))));
+
+        Checkpoint async = context.checkpoint();
+
+        // Disable CA generation
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editSpec()
+                    .editClusterCa()
+                        .withGenerateCertificateAuthority(false)
+                    .endClusterCa()
+                .endSpec()
                 .build();
+        NewMockCaReconciler mockCaReconciler = new NewMockCaReconciler(reconciliation, kafka, new ClusterOperatorConfig.ClusterOperatorConfigBuilder(ResourceUtils.dummyClusterOperatorConfig(), KafkaVersionTestUtils.getKafkaVersionLookup()).with(ClusterOperatorConfig.OPERATION_TIMEOUT_MS.key(), "1").build(),
+                supplier, vertx, CERT_MANAGER, PASSWORD_GENERATOR);
+        mockCaReconciler
+                .reconcile(Clock.systemUTC())
+                .onComplete(context.succeeding(c -> context.verify(() -> {
+                    assertThat("Kafka restart reasons", mockCaReconciler.kafkaRestartReasons, aMapWithSize(6));
+                    mockCaReconciler.kafkaRestartReasons.forEach((podName, restartReasons) -> {
+                        assertThat("Restart reasons for pod " + podName, restartReasons.getReasons(), hasSize(1));
+                        assertThat("Restart reasons for pod " + podName, restartReasons.contains(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED), is(true));
+                    });
+                    assertThat("Deployment restart reasons", mockCaReconciler.deploymentRestartReasons, aMapWithSize(3));
+                    mockCaReconciler.deploymentRestartReasons.forEach((deploymentName, restartReason) ->
+                            assertThat("Deployment restart reason for " + deploymentName, restartReason.equals(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED.getDefaultNote()), is(true)));
+                    async.flag();
+                })));
     }
 
-    public static Deployment deploymentWithName(String name) {
-        return new DeploymentBuilder()
+    @Test
```

Comment:
Do you think adding a small comment to the tests about why outcome of testStrimziManagedClusterCaCertRenewed and testUserManagedClusterCaCertRenewed being different?

Reply:
Do you think the comment on line 1935 is enough? 

```
// When user is managing CA a cert renewal implies a key replacement
```

Reply:
I think that is good enough